zebra-network/src/meta_addr.rs

```diff
@@ -458,6 +458,25 @@ impl MetaAddr {
         }
     }
 
+    /// Has this peer been seen recently?
+    ///
+    /// Returns `true` if this peer has responded recently or if the peer was gossiped with a
+    /// recent reported last seen time.
+    ///
+    /// [`constants::REACHABLE_PEER_DURATION`] represents the maximum time since a peer was seen to
+    /// still be considered reachable.
+    pub fn was_recently_reachable(&self) -> bool {
```

Comment:
Confusing method names - all these names are quite similar, but they do different things:

Rate-limiting reconnection attempts:
* `was_recently_live`
* `was_recently_attempted`
* `was_recently_failed`
* `LIVE_PEER_DURATION`

Gossiping addresses to other peers:
* `was_recently_reachable`
* `REACHABLE_PEER_DURATION`

Can we rename them based on what they are used for?
(`fastmod` is a really useful tool for mass renames.)

Also it might help to rename `is_ready_for_attempt`, or move it straight after the methods it calls.

Naming things is hard, let me know if you'd like to brainstorm the names here.

Reply:
Naming is hard indeed ☹️ Brainstorming could help 👍 . I considered changing `was_recently_reachable` to `can_be_gossiped`, but then I'm not sure what to name `REACHABLE_PEER_DURATION`.

As for the others, the only idea I managed to have was to call them something like `recently_responded`, `rate_limited_by_previous_attempt`, `rate_limited_by_previous_failure`, but I'm not sure if that helps much 🤔

Reply:
Here are some ideas:

Rate-limiting reconnection attempts - keyword: `connection`:
* `is_ready_for_attempt` -> `is_ready_for_connection_attempt`
* `was_recently_live` -> `connection_recently_responded`
* `was_recently_attempted` -> `connection_recently_attempted`
* `was_recently_failed` -> `connection_recently_failed`
* `LIVE_PEER_DURATION` -> `MIN_PEER_RECONNECTION_DELAY`

Gossiping addresses to other peers - keywords: `active` and `gossip`:
* `was_recently_reachable` -> `is_active_for_gossip`
* `REACHABLE_PEER_DURATION` -> `MAX_PEER_ACTIVE_FOR_GOSSIP`

Feel free to add is/was/has, and change the words, word order and grammar.

Reply:
Done, [spread across a few commits](https://github.com/jvff/zebra/compare/stop-gossiping-unreachable-addresses-r3..stop-gossiping-unreachable-addresses-r3-fixes). The only one I think became a bit weird was `has_connection_recently_responded`. It seems like `has_recently_responded` is more straight-forward, but maybe having `connection` in the name helps grouping the methods. We can reevaluate this again in the future if needed :+1: